src/server/modes/charts/plugins/datalens/preparers/pie/preparePie.ts

```diff
@@ -221,11 +252,57 @@ export function preparePie({
         });
     }
 
-    pie.data = mapAndColorizeGraphsByDimension({
-        graphs: pie.data,
-        colorsConfig,
-        isColorsItemExists: Boolean(color),
-    }) as (PiePoint & ExtendedSeriesLineOptions)[];
+    const isColoringByMeasure = color.type === 'MEASURE' && isNumberField(color);
+
+    if (isColoringByMeasure) {
+        pie.data = mapAndColorizePieByMeasure(pie.data, colorsConfig);
+    } else {
+        pie.data = mapAndColorizeGraphsByDimension({
+            graphs: pie.data,
+            colorsConfig,
+            isColorsItemExists: Boolean(color),
+        }) as (PiePoint & ExtendedSeriesLineOptions)[];
+    }
+
+    const customConfig: Record<string, any> = {};
+    const isLegendEnabled = shared.extraSettings?.legendMode !== 'hide';
+
+    if (isColoringByMeasure) {
+        pie.showInLegend = false;
+
+        const colorValues = pie.data.map((point) => Number(point.colorValue));
+        const points = pie.data as unknown as Highcharts.PointOptionsObject[];
+
+        const minColorValue = Math.min(...colorValues);
+        const maxColorValue = Math.max(...colorValues);
+
+        customConfig.colorAxis = {
+            startOnTick: false,
+            endOnTick: false,
+            min: minColorValue,
+            max: maxColorValue,
+            stops: getGradientStops(colorsConfig, points, minColorValue, maxColorValue),
+        };
+
+        customConfig.legend = {
+            title: {
+                text: getFakeTitleOrTitle(color),
+            },
+            enabled: isLegendEnabled,
+            symbolWidth: null,
+        };
+
+        customConfig.plotOptions = {
+            bar: {
+                borderWidth: 1,
+            },
+            column: {
+                borderWidth: 1,
+            },
+        };
+
+        ChartEditor.updateHighchartsConfig(customConfig);
```

Comment:
`customConfig` creation and `ChartEditor.updateHighchartsConfig` should be in the prepareHighchartsPie file - the config is prepared differently for highcharts and D3.